packages/flutter/lib/src/material/switch.dart

```diff
@@ -595,24 +609,24 @@ class Switch extends StatelessWidget {
 
     switch (_switchType) {
       case _SwitchType.material:
-        effectiveActiveThumbColor = activeColor;
+        effectiveActiveThumbColor = activeThumbColor;
       case _SwitchType.adaptive:
         switch (Theme.of(context).platform) {
           case TargetPlatform.android:
           case TargetPlatform.fuchsia:
           case TargetPlatform.linux:
           case TargetPlatform.windows:
-            effectiveActiveThumbColor = activeColor;
+            effectiveActiveThumbColor = activeThumbColor;
           case TargetPlatform.iOS:
           case TargetPlatform.macOS:
-            effectiveActiveTrackColor = activeColor;
+            effectiveActiveTrackColor = activeThumbColor;
```

Comment:
We can change here to be 
```suggestion
            effectiveActiveTrackColor = activeTrackColor ?? activeColor;
```
Then in line 630, it would be `activeTrackColor: effectiveActiveTrackColor`.

Or we don't need to change here at all.

Reply:
> I’m not quite sure why an intermediate step is needed here.

Previously, `Switch.adaptive` directly call `CupertinoSwitch` when it's on iOS and for `CupertinoSwitch.activeColor`, it is used for track color. So to not cause massive breaking changes, I added this conversion when I was removing `CupertinoSwitch` in `Switch.adaptive`.

Reply:
I understand now. Thank you for your explanation.